### PR TITLE
added vyper for vrf, price feeds, and keepers

### LINKED ISSUE #889: [Enhancement]: Vyper Support
### What would you like us to improve?

Add vyper support.

Now that the [apeworx-starter-kit](https://github.com/smartcontractkit/apeworx-starter-kit) is up, we can add vyper to our docs.

Vyper is the 2nd most popular smart contract programming language with about 8% of TVL written in vyper. 

https://defillama.com/languages

### How important it is?

Less important

### PR DESCRIPTION
**Pull request by PatrickAlphaC**
## Closing issues

closes #889

...

## Description

Added vyper support for VRF, Price Feeds, and Keepers 

I didn't add API support since I want to hold off on having to rewrite all the oracle contracts in vyper.

### CODE REVIEW

File: docs/Using Randomness/get-a-random-number.md [hunk 1]
```diff
@@ -160,3 +168,8 @@ After you are done with this contract and the subscription, you can retrieve the
 1. Under your subscription details, click **Cancel subscription**. A field opens asking which wallet address you want to send the remaining funds to.
 
 1. Enter your wallet address and click **Cancel subscription**. MetaMask opens and asks you to confirm the transaction. After you approve the transaction, Chainlink VRF closes your subscription account and sends the remaining LINK to your wallet.
+
+## Vyper Example
+
+You'd need to import the `VRFCoordinatorV2` vyper interface, you can find it  [here](https://github.com/smartcontractkit/apeworx-starter-kit/blob/main/contracts/interfaces/VRFCoordinatorV2.vy).
```
Comment (ritika-b) [inline docs/Using Randomness/get-a-random-number.md:174]:
```suggestion
You must import the `VRFCoordinatorV2` vyper interface. You can find it  [here](https://github.com/smartcontractkit/apeworx-starter-kit/blob/main/contracts/interfaces/VRFCoordinatorV2.vy).
```

Comment (ritika-b) [reply docs/Using Randomness/get-a-random-number.md:174]: Also, is Vyper captialized? I wasn't too sure

**Review by ritika-b — DISMISSED** [review]

File: docs/chainlink-keepers/compatible-contracts.md [hunk 2]
```diff
@@ -156,6 +155,12 @@ You can perform complex and broad off-chain computation, then execute on-chain s
 
 - **Identify the subset of states that must be updated**: If your contract maintains complicated objects such as arrays and structs, or stores a lot of data, you should read through your storage objects within your `checkUpkeep` and run your proprietary logic to determine if they require updates or maintenance. After that is complete, you can pass the known list of objects that require updates through the `performData` function.
 
+## Vyper Example
```
Comment (dwightjl) [inline docs/chainlink-keepers/compatible-contracts.md:158]:
```suggestion
## Vyper example
```

File: docs/chainlink-keepers/compatible-contracts.md [hunk 3]
```diff
@@ -17,30 +17,30 @@ Use custom logic to allow Keepers to determine when to execute your smart contra
...
+    - [`checkData`](#checkdata)
+    - [`performData`](#performdata)
+  - [`performUpkeep` function](#performupkeep-function)
+    - [`performData`](#performdata-1)
+- [Vyper Example](#vyper-example)
+- [Best practices](#best-practices)
+  - [Revalidate `performUpkeep`](#revalidate-performupkeep)
+  - [Test your contract](#test-your-contract)
 
 ## Example Contract
```
Comment (dwightjl) [inline docs/chainlink-keepers/compatible-contracts.md:33]:
```suggestion
## Example contract
```

File: docs/Using Randomness/get-a-random-number.md [hunk 4]
```diff
@@ -160,3 +168,8 @@ After you are done with this contract and the subscription, you can retrieve the
 1. Under your subscription details, click **Cancel subscription**. A field opens asking which wallet address you want to send the remaining funds to.
 
 1. Enter your wallet address and click **Cancel subscription**. MetaMask opens and asks you to confirm the transaction. After you approve the transaction, Chainlink VRF closes your subscription account and sends the remaining LINK to your wallet.
+
+## Vyper Example
```
Comment (dwightjl) [inline docs/Using Randomness/get-a-random-number.md:172]:
```suggestion
## Vyper example
```

**Review by dwightjl — CHANGES_REQUESTED** [review]
LGTM with a few minor changes.

**Comment by aelmanaa** [comment]
> LGTM with a few minor changes.

@dwightjl check now please

**Review by dwightjl — APPROVED** [review]

File: docs/Using Randomness/get-a-random-number.md (hunk 1 above)
Comment (aelmanaa) [reply docs/Using Randomness/get-a-random-number.md]: @ritika-b  please approve :)

**Pull request merged**